databricks/koalas/groupby.py

```diff
@@ -2031,28 +2037,26 @@ def __getattr__(self, item: str) -> Any:
         raise AttributeError(item)
 
     def _diff(self, *args, **kwargs):
-        groupkey_scols = [s._scol for s in self._groupkeys]
-        return Series._diff(self._kser, *args, **kwargs, part_cols=groupkey_scols)
+        return Series._diff(self._kser, *args, **kwargs, part_cols=self._groupkeys_scols)
 
     def _cum(self, func):
-        groupkey_scols = [s._scol for s in self._groupkeys]
-        return Series._cum(self._kser, func, True, part_cols=groupkey_scols)
+        return Series._cum(self._kser, func, True, part_cols=self._groupkeys_scols)
 
     def _rank(self, *args, **kwargs):
-        groupkey_scols = [s._scol for s in self._groupkeys]
-        return Series._rank(self._kser, *args, **kwargs, part_cols=groupkey_scols)
+        return Series._rank(self._kser, *args, **kwargs, part_cols=self._groupkeys_scols)
 
     def _fillna(self, *args, **kwargs):
-        groupkey_scols = [s._scol for s in self._groupkeys]
-        return Series._fillna(self._kser, *args, **kwargs, part_cols=groupkey_scols)
+        return Series._fillna(self._kser, *args, **kwargs, part_cols=self._groupkeys_scols)
 
     def _shift(self, periods, fill_value):
-        groupkey_scols = [s._scol for s in self._groupkeys]
-        return Series._shift(self._kser, periods, fill_value, part_cols=groupkey_scols)
+        return Series._shift(self._kser, periods, fill_value, part_cols=self._groupkeys_scols)
 
     @property
     def _kdf(self) -> DataFrame:
-        return self._kser._kdf
+        # TODO: if names from _kser and _groupkeys are name, grouping key is just ignored.
+        #    it can be a problem when both series have the same name but different operations.
+        series = [self._kser] + [s for s in self._groupkeys if s.name != self._kser.name]
```

Comment:
How about using `s._equals(self._kser)` which compares the `._scol._jc`s?

Reply:
Ah, thank you. Nice.

Reply:
In that case, seems it fails because the same names exist in the same DataFrame `AnalysisException: Reference 'b' is ambiguous, could be: b, b.;` ... but yeah I think it's better to fail fast for now rather than returning a wrong result.